lib/ruby_lsp/requests/hover.rb

```diff
@@ -95,6 +96,25 @@ def on_constant_path_node_enter(node)
         generate_hover(node.slice, node.location)
       end
 
+      sig { params(node: Prism::CallNode).void }
+      def on_call_node_enter(node)
+        return if DependencyDetector.instance.typechecker
+        return if node.receiver
+
+        message = node.message
+        return unless message
+
+        target_method = @index.resolve_method(message, @nesting.join("::"))
```

Comment:
I wonder if we can also try to resolve `new` to `initialize` here?

Reply:
Let's also wait for the singleton class support since it requires handling the receiver in the same way. I added an item to our checklist in #899 so that we do not forget.